Allow using Snowflake Connector for the actual DB Connector instead of the connection parameters.

sfc-gh-dkurokawa opened the pull request:
# Description
Allow using Snowflake Connector for the actual DB Connector instead of the connection parameters.

## Other details good to know for developers


## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Add support for using a Snowflake Connector object in `SnowflakeConnector` instead of connection parameters, with updates to tests and documentation.
> 
>   - **Behavior**:
>     - `SnowflakeConnector` in `connector.py` now accepts `snowpark_session` as an optional parameter, allowing the use of an existing Snowflake session instead of connection parameters.
>     - Raises `ValueError` if both `snowpark_session` and connection parameters are provided or if required parameters are missing.
>     - Updates `relevance()` in `llm_provider.py` to include `min_score_val` and `max_score_val` in `generate_system_prompt()` call.
>   - **Internal Methods**:
>     - Adds `_init_with_snowpark_session()` in `connector.py` to initialize with a Snowflake session.
>     - Removes `_create_snowflake_database_url()` and `_initialize_snowflake_server_side_feedback_evaluations()` from `connector.py`.
>   - **Tests**:
>     - Updates `test_snowflake_feedback_evaluation.py` to use `snowpark_session` in tests.
>     - Modifies `get_session()` in `snowflake_test_case.py` to pass `init_server_side=True`.
>   - **Documentation**:
>     - Fixes typo in `index.md` from `trulens-connectors-snowlake` to `trulens-connectors-snowflake`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for f703f4f45cec0e0923f1d977a9de90ee5b9e2b20. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

sfc-gh-pdharmana commented on src/connectors/snowflake/trulens/connectors/snowflake/connector.py:
```diff
@@ -34,30 +34,118 @@ class SnowflakeConnector(DBConnector):
     ):
-        database_args = database_args or {}
+        connection_parameters = {
+            "account": account,
+            "user": user,
+            "password": password,
+            "database": database,
+            "schema": schema,
+            "warehouse": warehouse,
```
Its okay if these are None? we could also make the keys only exist if values are none.

sfc-gh-dkurokawa replied:
so I check one of the following is true:
1. All set and `snowpark_session` is not set in lines 60-68.
2. None set and `snowpark_session` is set in lines 87-94.

sfc-gh-jreini reviewed (changes requested):
Can you update this in [docs](https://www.trulens.org/trulens/tracking/logging/where_to_log/log_in_snowflake/) and [notebook examples](https://github.com/truera/trulens/tree/main/examples/expositional/models/snowflake_cortex)

sfc-gh-dkurokawa commented:
> Can you update this in [docs](https://www.trulens.org/trulens/tracking/logging/where_to_log/log_in_snowflake/) and [notebook examples](https://github.com/truera/trulens/tree/main/examples/expositional/models/snowflake_cortex)

Okay I updated the docs. The notebook examples don't need to be updated as I just added functionality to connect another way. The original way still works.
